### PR TITLE
implementing the -indent option #270

### PR DESCRIPTION
**Pull request by snurmine**
Basic impl of intent. Hopefully I understood correctly this. Basically replacing default "  " with value from intent parameter.

### CODE REVIEW

File: cmd/eskip/args.go [hunk 1]
```diff
@@ -38,6 +39,7 @@ const (
 	appendFiltersFlag  = "append"
 	appendFileFlag     = "append-file"
 	prettyFlag         = "pretty"
+	intentStrFlag      = "intent"
```
Comment (hjacobs) [inline cmd/eskip/args.go]: "intent" should be "indent" or "indentation"

Comment (snurmine) [reply cmd/eskip/args.go]: Thanks. I fixed typo.

File: cmd/eskip/args.go [hunk 2]
```diff
@@ -204,6 +209,15 @@ func processFileArg() (*medium, error) {
 		path: nonFlagArgs[0]}, nil
 }
 
+// if pretty print then check that indent matches pattern
+func processIndentStr() error {
+	if pretty && !(regexp.MustCompile("^[\\s]*$").MatchString(indentStr)) {
+		return invalidIndentStr
+	}
+	return nil
+
```
Comment (szuecs) [inline cmd/eskip/args.go]: please drop the empty line

File: eskip/string.go [hunk 3]
```diff
@@ -8,7 +8,13 @@ import (
 	"strings"
 )
 
+type PrettyPrintInfo struct {
+	Pretty    bool
+	IndentStr string
+}
+
 func escape(s string, chars string) string {
+
```
Comment (szuecs) [inline eskip/string.go]: here also drop the empty line

**Comment by szuecs** [comment]
Thanks @snurmine !
Code looks fine. 

@aryszka / @lmineiro you had some interaction on this issue before. What do you think?

**Comment by szuecs** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Comment by snurmine** [comment]
Thanks. I squashed commit for cleaner history.

**Comment by szuecs** [comment]
@snurmine anything we can do, that you feel more confident to get this merged? 

A bit background information: we need 4 eyes from our project to merge it and the mentioned people, which should review had no time. Sometimes PRs are hanging for a couple of days.

**Comment by snurmine** [comment]
I apologize. I try to help Zalando more.

**Comment by szuecs** [comment]
@snurmine every contribution is welcome. :)

**Comment by aryszka** [comment]
:+1:

**Pull request closed without merging**